pkg/tagset/hash_generator_test.go

```diff
@@ -113,3 +115,85 @@ func genTags(count int, div int) ([]string, []string) {
 
 	return tags, uniq
 }
+
+func TestDedup2Small(t *testing.T) {
+	g := NewHashGenerator()
+
+	cases := []struct {
+		name             string
+		l, r, expL, expR string
+	}{
+		{"empty", "", "", "", ""},
+		{"empty l small", "", "foo", "", "foo"},
+		{"empty r small", "foo", "", "foo", ""},
+		{"small-1", "foo,foo,bar", "ook", "foo,bar", "ook"},
+		{"small-2", "foo,bar", "foo,ook", "foo,bar", "ook"},
+		{"small-3", "foo", "ook,foo,eek", "foo", "ook,eek"},
```

Comment:
Added. "small-6" is technically not small, since total number of tags is larger than 4 (this was originally testing only the nested-loop variant of deduplication used for small tagsets). I renamed it to "mid-1".